src/platform_impl/web/web_sys/fullscreen.rs

```diff
@@ -19,21 +26,65 @@ pub fn request_fullscreen(document: &Document, canvas: &HtmlCanvasElement) {
         #[wasm_bindgen(method, js_name = requestFullscreen)]
         fn request_fullscreen(this: &RequestFullscreen) -> Promise;
 
+        #[wasm_bindgen(method, js_name = requestFullscreen)]
+        fn request_fullscreen_with_options(
+            this: &RequestFullscreen,
+            options: &FullscreenOptions,
+        ) -> Promise;
+
         #[wasm_bindgen(method, js_name = webkitRequestFullscreen)]
         fn webkit_request_fullscreen(this: &RequestFullscreen);
+
+        type FullscreenOptions;
+
+        #[wasm_bindgen(method, setter, js_name = screen)]
+        fn set_screen(this: &FullscreenOptions, screen: &ScreenDetailed);
+    }
+
+    thread_local! {
+        static REJECT_HANDLER: Closure<dyn FnMut(JsValue)> = Closure::new(|error| {
+            console::error_1(&error);
+            error!("Failed to transition to full screen mode")
+        });
+    }
+
+    if is_fullscreen(document, canvas) {
+        return;
     }
 
     let canvas: &RequestFullscreen = canvas.unchecked_ref();
 
-    if has_fullscreen_api_support(canvas) {
-        thread_local! {
-            static REJECT_HANDLER: Closure<dyn FnMut(JsValue)> = Closure::new(|_| ());
-        }
-        REJECT_HANDLER.with(|handler| {
-            let _ = canvas.request_fullscreen().catch(handler);
-        });
-    } else {
-        canvas.webkit_request_fullscreen();
+    match fullscreen {
+        Fullscreen::Exclusive(_) => error!("Exclusive full screen mode is not supported"),
+        Fullscreen::Borderless(Some(monitor)) => {
+            if monitor::has_screen_details_support(window) {
+                if let Some(monitor) = monitor.detailed(main_thread) {
+                    let options: FullscreenOptions = Object::new().unchecked_into();
+                    options.set_screen(&monitor);
+                    REJECT_HANDLER.with(|handler| {
+                        let _ = canvas.request_fullscreen_with_options(&options).catch(handler);
+                    });
+                } else {
+                    error!(
+                        "Selecting a specific screen for fullscreen mode requires a detailed \
+                         screen. See `MonitorHandleExtWeb::is_detailed()`."
+                    )
+                }
+            } else {
+                error!(
+                    "Fullscreen mode selecting a specific screen is not supported by this browser"
+                )
+            }
```

Comment:
Would probably early return the outer else's error.

Reply:
Done.